core/objects.go

```diff
@@ -118,6 +119,10 @@ func (cr *CertificateRequest) UnmarshalJSON(data []byte) error {
 		return err
 	}
 
+	if len(raw.CSR) >= 10 && raw.CSR[8] == 2 && raw.CSR[9] == 0 {
```

Comment:
Can you add the explanation from https://github.com/letsencrypt/boulder/issues/1902#issuecomment-224420684 about why bytes 8 and 9 are examined here as a comment? I think it would be useful to have that context nearby when you're looking at this `if`.